kratos/tests/cpp_tests/geometries/test_coupling_geometry.cpp

```diff
@@ -0,0 +1,84 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License          BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "tests/cpp_tests/geometries/test_geometry.h"
+#include "geometries/triangle_2d_3.h"
+#include "geometries/coupling_geometry.h"
+
+namespace Kratos {
+    namespace Testing {
+
+        /// Factory functions
+        /** Generates a sample triangle2D3 with Point.
+         * Generates a point type right triangle with origin in the origin and leg size 1.
+         * @return  Pointer to a triangle2D3
+         */
+        Triangle2D3<Point>::Pointer GeneratePointTriangle() {
+            return Triangle2D3<Point>::Pointer(new Triangle2D3<Point>(
+                Point::Pointer(new Point(0.0, 0.0, 0.0)),
+                Point::Pointer(new Point(1.0, 0.0, 0.0)),
+                Point::Pointer(new Point(0.0, 1.0, 0.0))
+                ));
+        }
+
+        /** Generates a sample triangle2D3 with Node.
+         * @return  Pointer to a triangle2D3
+         */
+        Triangle2D3<Node<3>>::Pointer GenerateNodeTriangle() {
+            return Triangle2D3<Node<3>>::Pointer(new Triangle2D3<Node<3>>(
+                Node<3>::Pointer(new Node<3>(1, 1.0, 1.0, 0.0)),
+                Node<3>::Pointer(new Node<3>(2, 3.0, 0.5, 0.0)),
+                Node<3>::Pointer(new Node<3>(3, 2.5, 2.0, 0.0))
```

Comment:
try to avoid the usage of `new`, use `Kratos::make_shared` or (`Kratos::make_unique`) instead